src/lrf.rs

```diff
@@ -55,13 +63,257 @@ pub const SGRPROJ_PARAMS_EPS: [[u8; 2]; 1 << SGRPROJ_PARAMS_BITS] = [
+  
+  let mut work: [i32; MAX_SB_SIZE+7] = [0; MAX_SB_SIZE+7];
+  let vfilter: [i32; 7] = [ coeffs[0][0] as i32,
+                            coeffs[0][1] as i32,
+                            coeffs[0][2] as i32,
+                            128 - 2 * (coeffs[0][0] as i32 +
+                                       coeffs[0][1] as i32 +
+                                       coeffs[0][2] as i32 ),
+                            coeffs[0][2] as i32,
+                            coeffs[0][1] as i32,
+                            coeffs[0][0] as i32];
+  let hfilter: [i32; 7] = [ coeffs[1][0] as i32,
+                            coeffs[1][1] as i32,
+                            coeffs[1][2] as i32,
+                            128 - 2 * (coeffs[1][0] as i32 +
+                                       coeffs[1][1] as i32 +
+                                       coeffs[1][2] as i32),
+                            coeffs[1][2] as i32,
+                            coeffs[1][1] as i32,
+                            coeffs[1][0] as i32];
+
+  // unlike x, our y can be negative to start as the first stripe
+  // starts off the top of the frame by 8 pixels, and can also run off the end of the frame
+  let start_wi = if y < 0 {-y} else {0} as usize; 
+  let start_yi = if y < 0 {0} else {y} as usize; 
+  let end_i = cmp::max(0, if y+h > clipped_cfg.height as isize {
+    clipped_cfg.height as isize - y - start_wi as isize
+  } else {
+    h - start_wi as isize
+  }) as usize;
+  
+  let stride = out.cfg.stride;
+  let mut out_slice = out.mut_slice(&PlaneOffset{x: 0, y: start_yi as isize});
+  let out_data = out_slice.as_mut_slice();
+
+  for xi in x..x+w {
+    let n = cmp::min(7, clipped_cfg.width as isize + 3 - xi as isize);
+    for yi in y-3..y+h+4 {
+      let mut src_plane: &Plane;
+      let mut acc = 0;
+      let ly;
+      if yi < y {
+        ly = cmp::max(clamp(yi, 0, clipped_cfg.height as isize - 1), y - 2) as usize;
+        src_plane = deblocked;
+      } else if yi < y+h {
+        ly = clamp(yi, 0, clipped_cfg.height as isize - 1) as usize;
+        src_plane = cdeffed;
+      } else {
+        ly = cmp::min(clamp(yi, 0, clipped_cfg.height as isize - 1), y+h+1) as usize;
+        src_plane = deblocked;
+      }
+      
+      for i in 0..3 - xi as isize {
+        acc += hfilter[i as usize] * src_plane.p(0, ly) as i32;
+      }
+      for i in cmp::max(0,3 - (xi as isize))..n {
+        acc += hfilter[i as usize] * src_plane.p((xi as isize + i - 3) as usize, ly) as i32;
+      }
+      for i in n..7 {
+        acc += hfilter[i as usize] * src_plane.p(clipped_cfg.width - 1, ly) as i32;
+      }
+        
+      acc = acc + (1 << round_h >> 1) >> round_h;
+      work[(yi-y+3) as usize] = clamp(acc, -offset, limit-offset);
+    }
+
+    for (wi, dst) in (start_wi..start_wi+end_i).zip(out_data[xi..].iter_mut().step_by(stride).take(end_i)) {
+      let mut acc = 0;
+      for (i,src) in (0..7).zip(work[wi..wi+7].iter_mut()) {
```

Comment:
Maybe
``` rust
for (vf, &src) in vfilter[0..7].iter().zip(work[wi..wi+7].iter()) {
  acc += vf * src;
}
```

Reply:
Oh, ha, yeah.  Will simplify in an upcoming patch :-)